### DISCUSSION TITLE
Undefined "global" variable

### DISCUSSION THREAD

**Discussion by jessapp** [remix-run/remix#1657] [Q&A] [answered]
# Problem
I need to use [react-live](https://github.com/FormidableLabs/react-live) in my Remix app. When I try to use their components, I get the following error: 

![Screen Shot 2022-01-25 at 2 51 54 PM](https://user-images.githubusercontent.com/18542356/151073337-c58e9268-2cd5-46e1-b67d-57eb4595759b.png)

See [this repository](https://github.com/jessapp/remix-react-live-poc) for a reproduction (go to [this file](https://github.com/jessapp/remix-react-live-poc/blob/master/app/routes/index.tsx) to see the `react-live` code). 

The stack trace in my browser console leads me back to `react-simple-code-editor`, which is used under the hood by `react-live`. It seems that `react-simple-code-editor` expects the variable `global` to be defined, but it is not defined within the context of my Remix app.  

![Screen Shot 2022-01-25 at 2 52 00 PM](https://user-images.githubusercontent.com/18542356/151073343-26571aca-bde9-4f88-9ef0-02843f97e493.png)
![Screen Shot 2022-01-25 at 2 52 07 PM](https://user-images.githubusercontent.com/18542356/151073506-e56d02a1-ed2d-4e4e-87dd-73bcaa8ae52b.png)
![Screen Shot 2022-01-25 at 2 52 14 PM](https://user-images.githubusercontent.com/18542356/151073508-036ac932-6cbb-4b52-9b67-2c4034ea7878.png)

I'm not totally clear on why `global` is not defined, but I believe it could possibly have something to do with Node Javascript vs browser Javascript. 

# Solutions I have tried

Because I suspect this may have something to do with Node JS vs browser JS, I tried to install `@remix-run/node`, as per [this discussion](https://github.com/remix-run/remix/discussions/894). This didn't fix the issue. 

Does anybody know why `global` is undefined, and how I can move past this error? Thank you!

**Comment by sergiodxa** [comment, marked as answer]
The global object is a Node only thing. It doesn’t exists on the browser.

I think webpack does change global to window or globalThis, but since Remix doesn’t use webpack that doesn’t work, and most likely that package was coded expecting webpack specific and non standard behavior.

I would open an issue on their repository to ask them to use window or globalThis.
